src/ui/confirm-ui-handler.ts

```diff
@@ -1,123 +1,46 @@
-import type { OptionSelectConfig } from "./abstact-option-select-ui-handler";
-import AbstractOptionSelectUiHandler from "./abstact-option-select-ui-handler";
-import { Mode } from "./ui";
+import type { ConfirmModeConfig } from "#app/ui/interfaces/confirm-menu-config";
+import type { OptionSelectModeConfig, OptionSelectItem } from "#app/ui/interfaces/option-select-config";
+import OptionSelectUiHandler from "#app/ui/option-select-ui-handler";
+import { Mode } from "#app/ui/ui";
 import i18next from "i18next";
-import { Button } from "#enums/buttons";
-import { globalScene } from "#app/global-scene";
-
-export default class ConfirmUiHandler extends AbstractOptionSelectUiHandler {
-  public static readonly windowWidth: number = 48;
-
-  private switchCheck: boolean;
-  private switchCheckCursor: number;
 
+/**
+ * Handler that displays a simple Yes/No menu.
+ * @extends OptionSelectUiHandler
+ */
+export default class ConfirmUiHandler extends OptionSelectUiHandler {
   constructor() {
     super(Mode.CONFIRM);
   }
 
-  getWindowWidth(): number {
-    return ConfirmUiHandler.windowWidth;
-  }
-
   override show(args: any[]): boolean {
-    if (
-      args.length === 4
-      && args[0] instanceof Function
-      && args[1] instanceof Function
-      && args[2] instanceof Function
-      && args[3] === "fullParty"
-    ) {
-      const config: OptionSelectConfig = {
-        options: [
-          {
-            label: i18next.t("partyUiHandler:SUMMARY"),
-            handler: () => {
-              args[0]();
-              return true;
-            },
-          },
-          {
-            label: i18next.t("menu:yes"),
-            handler: () => {
-              args[1]();
-              return true;
-            },
-          },
-          {
-            label: i18next.t("menu:no"),
-            handler: () => {
-              args[2]();
-              return true;
-            },
-          },
-        ],
-        delay: args.length >= 8 && args[7] !== null ? (args[7] as number) : 0,
-      };
-
-      super.show([config]);
-
-      this.switchCheck = args.length >= 5 && args[4] !== null && (args[4] as boolean);
-
-      const xOffset = args.length >= 6 && args[5] !== null ? (args[5] as number) : 0;
-      const yOffset = args.length >= 7 && args[6] !== null ? (args[6] as number) : 0;
-
-      this.optionSelectContainer.setPosition(globalScene.game.canvas.width / 6 - 1 + xOffset, -48 + yOffset);
+    if (!args[0] || !args[0].hasOwnProperty("yesHandler") || !args[0].hasOwnProperty("noHandler")) {
```

Comment:
Please avoid using `hasOwnProperty()` the ts compiler can't handle it well
```suggestion
    if (!args[0] || !args[0].yesHandler || !args[0].noHandler) {
```